rasa/shared/core/training_data/structures.py

```diff
@@ -221,23 +218,85 @@ def explicit_events(
 
         return events
 
+    def get_rules_condition(self) -> Optional[List[Event]]:
+        pass
+
+    def get_rules_steps(self) -> Optional[List[Event]]:
+        pass
+
     def __repr__(self) -> Text:
         return (
             "StoryStep("
             "block_name={!r}, "
             "start_checkpoints={!r}, "
             "end_checkpoints={!r}, "
-            "is_rule={!r}, "
             "events={!r})".format(
                 self.block_name,
                 self.start_checkpoints,
                 self.end_checkpoints,
-                self.is_rule,
                 self.events,
             )
         )
 
 
+class RuleStep(StoryStep):
+    """A Special type of StoryStep representing a Rule. """
+
+    def __init__(
+        self,
+        block_name: Optional[Text] = None,
+        start_checkpoints: Optional[List[Checkpoint]] = None,
+        end_checkpoints: Optional[List[Checkpoint]] = None,
+        events: Optional[List[Union[Event, List[Event]]]] = None,
+        source_name: Optional[Text] = None,
+    ) -> None:
+        super().__init__(
+            block_name, start_checkpoints, end_checkpoints, events, source_name
+        )
+        self.condition_events_indices = set()
+
+    def __repr__(self) -> Text:
+        return (
+            "RuleStep("
+            "block_name={!r}, "
+            "start_checkpoints={!r}, "
+            "end_checkpoints={!r}, "
+            "events={!r})".format(
+                self.block_name,
+                self.start_checkpoints,
+                self.end_checkpoints,
+                self.events,
+            )
+        )
+
+    def get_rules_condition(self) -> List[Event]:
```

Comment:
The return type should match the parent's (`Optional`).

Reply:
removed.